app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -333,4 +337,34 @@ public function getEstimateCacheIdentifier($quote, $addressData)
 
         return md5($cacheIdentifier);
     }
+
+    /**
+     * This function checks for all special cases in which address validation should not be
+     * performed before making shipping and tax estimates.
+     *
+     * While we currently only have one case, redacted address information from an Apple Pay request,
+     * this wrapper function is justified because there maybe future payment additions that do similar
+     * address data redaction like Google Pay, Samsung Pay, Paypal, etc.
+     */
+    protected function shouldDoAddressValidation() {
+        return !($this->isApplePayRequest());
+    }
+
+    /**
+     * Checks whether this is an Apple Pay request.  Currently, Apple Pay request are populated with
+     * "n/a" in several required address fields, particularly the "name" field, which is Bolt defined,
+     *  not customer defined.  Additionally, the "phone" field will be null.
+     *
+     * Standard Bolt request leave the "name" field as null while forcing the user to populate
+     * the phone field, and therefore is never null.
+     *
+     * We do anticipate Bolt server-side refinement for indicating Apple Pay request, likely via User-Agent
+     * or a custom HTTP request header.  For now, we'll rely on sentinel value detection.
+     */
+    private function isApplePayRequest() {
+        $requestData = json_decode($this->_requestJSON);
+        $shippingAddress = $requestData->shipping_address;
+
+        return ($shippingAddress->name === 'n/a') && is_null($shippingAddress->phone);
```

Comment:
Might just get away with `($shippingAddress->name === 'n/a')`.

Reply:
@chaseadams509 
probably.  In the case that ApplePay may start sending the phone, I will remove the more strict check.  The important thing was me to make you aware that the 'n/a' in the name field is what is being used by this plugin because it can not be set by the customer.